parlai/agents/language_model/language_model.py

```diff
@@ -132,8 +135,10 @@ def __init__(self, opt, shared=None):
                 # load model parameters if available
                 print('Loading existing model params from ' + init_model)
                 new_opt, self.states = self.load(init_model)
-                # override model-specific options with stored ones
-                opt = self.override_opt(new_opt)
+                # override model-specific options with stored ones if not
```

Comment:
i think the code at 126 might cause a problem because parlai load agent will pass in the saved opts, which may include an init_model and a model_file option, and it will load from the init_model rather than model_file..  shouldnt it load from model_file if it exists (overrides 'init_model')?

Reply:
good catch... i thought we wanted to load from init_model if it exists (and save into model_file). if that's the case, we should check for an init_model in `load_agent_module` in `parlai/core/agents.py`. but if it should be the other way around, why do we have init_model? what do you think?

Reply:
i think when you begin training you might start from an init_model, but then after you have a model_file you would use that instead.  i'm not sure if this is model specific code, or if we want  to migrate that agents.py

i've tried to make a PR for seq2seq here, note some other changes with dict also: https://github.com/facebookresearch/ParlAI/pull/739/files